CMakeLists.txt

```diff
@@ -1,33 +1,68 @@
-cmake_minimum_required(VERSION 3.5.1)
-project(arbitration_graphs)
+cmake_minimum_required(VERSION 3.22)
 
-set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+######################
+## Project settings ##
+######################
+
+# Read the project/package version from ./version file
+file(READ "${CMAKE_CURRENT_SOURCE_DIR}/version" ver)
+string(REGEX MATCH "VERSION=v([0-9\.]*)" _ ${ver})
+set(PACKAGE_VERSION ${CMAKE_MATCH_1})
+
+# Setup project
+project(arbitration_graphs
+  VERSION ${PACKAGE_VERSION}
+  DESCRIPTION "Hierarchical behavior models for complex decision-making and behavior generation in robotics"
+  LANGUAGES CXX
+)
+
+# Select project features
+set(BUILD_TESTS "FALSE" CACHE STRING "Compile the project with all unit tests")
+
+
+###############
+## C++ setup ##
+###############
+
+# Only do these if this is the main project, and not if it is included through add_subdirectory
+if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+  # Require C++17
+  set(CMAKE_CXX_STANDARD 17)
```

Comment:
What if the package including this package sets the cxx standard to e.g. 11?

Reply:
Then, compilation will fail. C++17 is the minimum we require.
Such users can install our package before usage.

But, with this formulation, we allow users to include the package as subdirectory when using newer standards.